src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/Transfer.java

```diff
@@ -93,4 +87,25 @@ public Optional<RaptorTransfer> asRaptorTransfer(StreetSearchRequest request) {
       )
     );
   }
+
+  public static Optional<State> miniAstar(final State s, Collection<Edge> edges) {
```

Comment:
We need to think about this name, I'd say something like `traverseEdges` might sound good. Also it might make sense to try to traverse all generated states, since one of the returned states might return null, when traversing the next edge.